cpp/src/search/contains_table_impl.cu

```diff
@@ -0,0 +1,128 @@
+  if (nullable_columns.size() > 1) {
+    auto row_bitmask =
+      cudf::detail::bitmask_and(
+        table_view{nullable_columns}, stream, cudf::get_current_device_resource_ref())
+        .first;
+    auto const row_bitmask_ptr = static_cast<bitmask_type const*>(row_bitmask.data());
+    return std::pair(std::move(row_bitmask), row_bitmask_ptr);
+  }
+
+  return std::pair(rmm::device_buffer{0, stream}, nullable_columns.front().null_mask());
+}
+
+// Explicit instantiations for non-nested types (HasNested=false)
+using experimental_hasher_adapter = hasher_adapter<
+  cudf::experimental::row::hash::device_row_hasher<cudf::hashing::detail::default_hash,
+                                                   nullate::DYNAMIC>,
+  cudf::experimental::row::hash::device_row_hasher<cudf::hashing::detail::default_hash,
+                                                   nullate::DYNAMIC>>;
+
+template void dispatch_nan_comparator<false, experimental_hasher_adapter>(
+  table_view const& haystack,
+  table_view const& needles,
+  null_equality compare_nulls,
+  nan_equality compare_nans,
+  bool haystack_has_nulls,
+  bool needles_has_nulls,
+  bool has_any_nulls,
+  cudf::experimental::row::equality::self_comparator self_equal,
+  cudf::experimental::row::equality::two_table_comparator two_table_equal,
+  experimental_hasher_adapter const& d_hasher,
+  rmm::device_uvector<bool>& contained,
+  rmm::cuda_stream_view stream);
+
+// For HasNested=false (non-nested columns) with nan_equal_comparator
+using nan_equal_self_comparator = cudf::experimental::row::equality::device_row_comparator<
+  false,
+  cudf::nullate::DYNAMIC,
+  cudf::experimental::row::equality::nan_equal_physical_equality_comparator>;
+
+using nan_equal_two_table_comparator =
+  cudf::experimental::row::equality::strong_index_comparator_adapter<nan_equal_self_comparator>;
+
+using nan_equal_comparator_adapter =
+  comparator_adapter<nan_equal_self_comparator, nan_equal_two_table_comparator>;
+
+template void perform_contains(
+  table_view const& haystack,
+  table_view const& needles,
+  bool haystack_has_nulls,
+  bool needles_has_nulls,
+  null_equality compare_nulls,
+  nan_equal_comparator_adapter const& d_equal,
+  cuco::linear_probing<1, experimental_hasher_adapter> const& probing_scheme,
+  rmm::device_uvector<bool>& contained,
+  rmm::cuda_stream_view stream);
+
+// For HasNested=false (non-nested columns) with nan_unequal_comparator
+using nan_unequal_self_comparator = cudf::experimental::row::equality::device_row_comparator<
+  false,
+  cudf::nullate::DYNAMIC,
+  cudf::experimental::row::equality::physical_equality_comparator>;
+
+using nan_unequal_two_table_comparator =
+  cudf::experimental::row::equality::strong_index_comparator_adapter<nan_unequal_self_comparator>;
+
+using nan_unequal_comparator_adapter =
+  comparator_adapter<nan_unequal_self_comparator, nan_unequal_two_table_comparator>;
+
+template void perform_contains(
```

Comment:
Nit: It should be `check_contains` instead of `perform_` as `perform_` doesn't make sense.